### ISSUE TITLE
Tree and Tree Item - Ensure all interactive functionality is operable with the keyboard - (2036570420)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5617]
### Module: 
 38 Tree and Tree Item 

### Violation: 
 Ensure all interactive functionality is operable with the keyboard 

### WCAG Reference: 
- [2.1.1 Keyboard](https://www.w3.org/WAI/WCAG21/Understanding/keyboard.html) 

### Instance ID: 
 2036570420 

### Severity: 
 8 

### Description: 
 [Issue]
There are expand/collapse controls that cannot be operated with the keyboard alone. Examples include:

- Small
- Medium
- Large
- Test 2. Tree Lines: For visual testing only. Adds lines between related tree items
- Single
- Multi
- Ancestors
- Test 4. Expanded Tree item
- Test 5. Selected Tree item

[User Impact]

Keyboard-only users will be prevented from accessing the functionality provided by these controls.

[Code Reference] 
 ```
<calcite-tree-item aria-expanded="false" aria-hidden="false" role="treeitem" tabindex="0" depth="1" has-children="" scale="s" calcite-hydrated="">
  Child 2
  <calcite-tree slot="children" tabindex="-1" child="" scale="s" selection-mode="single" calcite-hydrated="">
    (...)
  </calcite-tree>
</calcite-tree-item>
 ``` 

### Note: 
 [Suggestion]
Ensure all interactive functionality is operable with the keyboard. The best way to accomplish this is by using appropriate native controls, which come with keyboard functionality built in.

If using a native control is not possible, the control must have JavaScript event listeners that allow the control to be activated when Enter is pressed.

[Compliant Code Example] 
 ```
<calcite-tree-item aria-expanded="false" aria-hidden="false" role="treeitem" tabindex="0" depth="1" has-children="" scale="s" calcite-hydrated="">
  Child 2
  <calcite-tree slot="children" tabindex="-1" child="" scale="s" selection-mode="single" calcite-hydrated="">
    (...)
  </calcite-tree>
</calcite-tree-item>
 ```
 

### Media Type: 
 Keyboard Accessibility 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
Duplicate of https://github.com/Esri/calcite-components/issues/5617
